app/lib/services/background_service.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:background_fetch/background_fetch.dart';
+import 'package:threebotlogin/services/nodes_check_service.dart';
+import 'notification_service.dart';
+
+void backgroundFetchHeadlessTask(HeadlessTask task) async {
+  final String taskId = task.taskId;
+  final bool timeout = task.timeout;
+
+  if (timeout) {
+    BackgroundFetch.finish(taskId);
+    return;
+  }
+  await checkNodeStatus();
+
+  BackgroundFetch.finish(taskId);
+}
+
+Future<void> checkNodeStatus() async {
+  final offlineNodes = await NodeCheckService.pingNodesInBackground();
+  if (offlineNodes.isEmpty) return;
+
+  final now = DateTime.now().millisecondsSinceEpoch;
+  final sevenDaysAgoTimestamp =
+      DateTime.now().subtract(const Duration(days: 7)).millisecondsSinceEpoch;
+
+  final nodesToNotify = offlineNodes.where((node) {
+    if (node.updatedAt! <= sevenDaysAgoTimestamp) return false;
+
+    final downtime = Duration(milliseconds: now - node.updatedAt!);
```

Comment:
The updatedAt is in seconds, so that the equation will result in the wrong values